### DISCUSSION TITLE
Utilizing the AI Deck bootloader for building user code

### DISCUSSION THREAD

**Discussion by dielso** [bitcraze/discussions#243] [Q&A] [answered]
Hello!

I've written some code for the AI Deck before getting my hands on it, so I've only built and ran it on gvsoc. Now that I have it, I don't have the JTAG connector. Fortunately, I was able to flash the examples via wi-fi. Still, I have no idea on how to generate a binary (such as the .bin for the wifi streaming example) of a code. I'm guessing I should be able to do it with the [AI Deck GAP8 bootloader](https://github.com/bitcraze/aideck-gap8-bootloader) and the [toolbelt](https://github.com/bitcraze/toolbelt), but I'm not sure on how to do it and am too afraid of messing the board by testing it like crazy.
So, I would be very happy if someone had a (safe) method on it.

Thanks!

**Comment by knmcguire** [comment, marked as answer]
Hi! We have a getting started tutorial here: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/

Once you have gone through that, you could try to build and flash the wifi example over air with a crazyradio: https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/test-functions/wifi-streamer/

**Reply by dielso** [reply]
Firstly, thanks for the reply. I had already done the wifi-streamer example in the getting started and now I built it with the docker instructions.
Unfortunately I didn't understand quite well how I would apply the "tools/build/make-example" on another code, since I didn't get it to build on the other examples (such as test_camera, image_manipulations and simple_kernel examples).
I have built them and my own code with:
`docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck:4.8.0.1 /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make clean all image'`
But the binary output stops flashing at 0% :laughing: 
Any help is very much welcome!

**Reply by evoggy** [reply]
Hi!

The flashing progress shouldn't depend on how you've built the binary, it will flash anything.

How are you flashing the deck? Via cfloader or via WiFi?

Did you change any other firmware on the Crazyflie between successfully flashing the examples and now not being able to flash?

**Reply by dielso** [reply]
Hello!

I'm flashing with cfloader.The flashing was stopping at 0% because I had both the AI Deck and the Flow Deck mounted. That messed up things a bit, but I simply ran the cfclient bootloader and it worked fine again.
By building with the command above and flashing the binary with the command below (in the video streamer example) I'm being able to program it without the connector, as I wanted.
Thanks!

`cfloader flash path/to/BUILD/GAP8_V2/GCC_RISCV_[OS]/target.board.devices.flash.img deck-bcAI:gap8-fw -w radio://0/80/2M`

**Reply by evoggy** [reply]
Great, good to hear that it's working! What you're describing is unfortunately a bug ([link](https://github.com/bitcraze/crazyflie-clients-python/issues/616)) that's not been fixed yet.
